sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala

```diff
@@ -542,6 +550,44 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
         }
         checkPushedInfo(df8, expectedPlanFragment8)
         checkAnswer(df8, Seq(Row(2, "david", 10000, 1300, true)))
+
+        val df9 = sql("SELECT name FROM h2.test.datetime WHERE " +
+          "dayofyear(date1) > 100 and dayofmonth(date1) > 10 and dayofweek(date1) > 1")
+        checkFiltersRemoved(df9)
+        val expectedPlanFragment9 =
+          "PushedFilters: [DATE1 IS NOT NULL, DAY_OF_YEAR(DATE1) > 100, " +
+          "DAY_OF_MONTH(DATE1) > 10, DAY_OF_WEEK(DATE1) > 1]"
+        checkPushedInfo(df9, expectedPlanFragment9)
+        checkAnswer(df9, Seq(Row("amy"), Row("alex")))
+
+        val df10 = sql("SELECT name FROM h2.test.datetime WHERE " +
+          "year(date1) = 2022 and quarter(date1) = 2 and month(date1) = 5")
+        checkFiltersRemoved(df10)
+        val expectedPlanFragment10 =
+          "PushedFilters: [DATE1 IS NOT NULL, YEAR(DATE1) = 2022, " +
+          "QUARTER(DATE1) = 2, MONTH(DATE1) = 5]"
+        checkPushedInfo(df10, expectedPlanFragment10)
+        checkAnswer(df10, Seq(Row("amy"), Row("alex")))
+
+        val df11 = sql("SELECT name FROM h2.test.datetime WHERE " +
+          "hour(time1) = 0 and minute(time1) = 0 and second(time1) = 0")
+        checkFiltersRemoved(df11)
+        val expectedPlanFragment11 =
+          "PushedFilters: [TIME1 IS NOT NULL, HOUR(TIME1) = 0, " +
+          "MINUTE(TIME1) = 0, SECOND(TIME1) = 0]"
+        checkPushedInfo(df11, expectedPlanFragment11)
+        checkAnswer(df11, Seq(Row("amy"), Row("alex")))
+
+        // H2 does not support
+        val df12 = sql("SELECT name FROM h2.test.datetime WHERE " +
+          "trunc(date1, 'week') = date'2022-05-16' and date_add(date1, 1) = date'2022-05-20' " +
+          "and datediff(date1, '2022-05-10') > 0 and extract(week from date1) = 20 " +
+          "and extract(YEAROFWEEK from date1) = 2022")
```

Comment:
Could we add tests just test `extract(week from date1)` and `extract(YEAROFWEEK from date1)`. We can see `week` and `YEAROFWEEK` will be pushed.